mixing/models/mixllama/modeling_mixllama.py

```diff
@@ -0,0 +1,462 @@
+import warnings
+from typing import List, Optional, Tuple, Union
+from functools import partial
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss
+import torch.nn.functional as F
+
+from transformers.models.llama.modeling_llama import (
+    LlamaPreTrainedModel,
+    LlamaModel,
+    LlamaForCausalLM,
+    LlamaDecoderLayer,
+    LlamaMLP,
+    LlamaRMSNorm,
+    LLAMA_ATTENTION_CLASSES
+)
+from transformers.modeling_outputs import (
+    MoeModelOutputWithPast,
+    MoeCausalLMOutputWithPast,
+)
+from transformers.cache_utils import Cache, StaticCache, DynamicCache
+from transformers.utils import logging
+
+from .configuration_mixllama import MixLlamaConfig
+from mixing.utils.loss import load_balancing as load_balancing_loss_func
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "MixLlamaConfig"
+
+def make_moe_modules(moe_class_partial):
```

Comment:
I don't think this extra level of indirection is required and using partials makes the code more confusing